projects/cds/src/merchandising/facade/cds-merchandising-product.service.ts

```diff
@@ -1,17 +1,42 @@
 import { Injectable } from '@angular/core';
-import { Observable } from 'rxjs';
+import { BaseSiteService, LanguageService } from '@spartacus/core';
+import { combineLatest, Observable } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+import { StrategyRequest } from './../../cds-models/cds-strategy-request.model';
 import { MerchandisingStrategyConnector } from './../connectors/strategy/merchandising-strategy.connector';
 import { MerchandisingProducts } from './../model/merchandising.products.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CdsMerchandisingProductService {
-  constructor(protected strategyConnector: MerchandisingStrategyConnector) {}
+  constructor(
+    protected strategyConnector: MerchandisingStrategyConnector,
+    protected baseSiteService: BaseSiteService,
+    protected languageService: LanguageService
+  ) {}
 
   loadProductsForStrategy(
-    strategyId: string
+    strategyId: string,
+    numberToDisplay?: number
   ): Observable<MerchandisingProducts> {
-    return this.strategyConnector.loadProductsForStrategy(strategyId);
+    // Logic here to get additional context, i.e. base site, language, etc
+
+    return combineLatest([
+      this.baseSiteService.getActive(),
+      this.languageService.getActive(),
+    ]).pipe(
+      map(([baseSite, language]: [string, string]) => {
+        const strategyRequest: StrategyRequest = {
+          site: baseSite,
+          language: language,
```

Comment:
In TS, if the property name matches the variable name, you just have one:
```ts
{
site,
language,
pageSize: numberToDisplay
}
```

Reply:
Cool, good tip, and done